### PR TITLE
Pull api clone/disconnect into sd-668

### PR DESCRIPTION
**Pull request by jacogr**
Pull changes from #681 into #684

### CODE REVIEW

File: packages/app-js/src/index.tsx [hunk 1]
```diff
@@ -55,22 +73,34 @@ class App extends React.PureComponent<Props, State> {
...
+                isCircular
+                isDisabled={!isRunning}
+                isNegative
+                icon='close'
+                onClick={this.stopJs}
+              />
+            </div>
           </Editor>
-          <Output className='js--Output' logs={logs}>
+          <Output logs={logs}>
```
Comment (Stefie) [inline packages/app-js/src/index.tsx]: `className='js--Output'` got lost here. Some of the styling is relying on this class.

File: packages/app-js/src/index.tsx [hunk 2]
```diff
@@ -102,15 +135,24 @@ class App extends React.PureComponent<Props, State> {
...
 
-    const exec = `(async ({${Object.keys(injected).join(',')}}) => {
-      try {
-        ${code}
-      } catch (error) {
-        console.error(error);
-      }
-    })(injected);`;
+    // squash into a single line so exceptions (with linenumbers) maps to the same line/origin
+    // as we have in the editor view (TODO: Make the console.error here actually return the full stack)
```
Comment (Stefie) [inline packages/app-js/src/index.tsx]: Nice, haven't thought of that!

File: packages/app-js/src/index.tsx (hunk 1 above)
Comment (jacogr) [reply packages/app-js/src/index.tsx]: Ahh, it is just removing duplication - it is applied in the `Output` (and `Editor`) component already, so the above was actually a duplication.

File: packages/app-js/src/index.tsx (hunk 2 above)
Comment (jacogr) [reply packages/app-js/src/index.tsx]: This happens when you think of it, squash it into a single line and never comment on it. My bad.

File: packages/app-js/src/index.tsx (hunk 1 above)
Comment (Stefie) [reply packages/app-js/src/index.tsx]: Ah, ok, closing!

**Pull request merged**